cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerModificationService.java

```diff
@@ -170,47 +174,114 @@ public void cleanupCluster(Telemetry telemetry) throws CloudbreakException {
     @Override
     public void upgradeClusterRuntime(Set<ClusterComponent> components, boolean patchUpgrade) throws CloudbreakException {
         try {
-            ClusterComponent stackComponent = getStackComponent(components);
-
-            ClouderaManagerProduct stackProduct = stackComponent.getAttributes().get(ClouderaManagerProduct.class);
-            String stackProductVersion = stackProduct.getVersion();
-            String stackProductParcel = stackProduct.getParcel();
-            String product = com.sequenceiq.cloudbreak.cloud.model.component.StackType.CDH.name();
-
             ClustersResourceApi clustersResourceApi = clouderaManagerApiFactory.getClustersResourceApi(apiClient);
-            ParcelsResourceApi parcelsResourceApi = clouderaManagerApiFactory.getParcelsResourceApi(apiClient);
             ParcelResourceApi parcelResourceApi = clouderaManagerApiFactory.getParcelResourceApi(apiClient);
             MgmtServiceResourceApi mgmtServiceResourceApi = clouderaManagerApiFactory.getMgmtServiceResourceApi(apiClient);
             ClouderaManagerResourceApi clouderaManagerResourceApi = clouderaManagerApiFactory.getClouderaManagerResourceApi(apiClient);
 
-            startClouderaManager(stack, apiClient);
+            startClouderaManager();
             checkParcelApiAvailability();
-            setParcelRepo(stackProductParcel, clouderaManagerResourceApi);
+
+            Set<ClouderaManagerProduct> services = getServices(components);
```

Comment:
Unfortunately, using the components is not going to work now. In the case of Data Lake, it will try to re-activate all parcels that are on the images and that doesn't work.